code/game/gamemodes/umbrage/umbrage.dm

```diff
@@ -0,0 +1,184 @@
+	thralls => veils
+	ascendants => progenitors
+
+Folder contents:
+	/code/game/gamemodes/umbrage
+		/__umbrage_defines.dm: Contains defines and global variables for all umbrage code.
+		/umbrage.dm: This file. Contains the skeleton of the gamemode.
+		/umbrage_antags.dm: Contains antagonist datums for umbrages and veils.
+		/umbrage_datum.dm: Contains the tracking datum for lots of umbrage stuff.
+		/umbrage_major_abilities.dm: Contains hatching and ascension abilities and their related objects.
+		/umbrage_minor_abilities.dm: Contains spells, abilities, and other stuff. Does NOT including hatching and ascension.
+		/umbrage_objects.dm: Contains the few items and structures used in Umbrage.
+		/umbrage_unsorted.dm: Contains things with nowhere else to be, like the umbrage race.
+
+Idea and initial code by Xhuis (my 3rd gamemode now...)
+
+*/
+
+///////////////
+// FRAMEWORK //
+///////////////
+
+/datum/game_mode
+	var/list/umbrages = list() //A list of the minds of all umbrages, including progenitors.
+	var/list/veils = list() //A list of the minds of all veils.
+	var/list/umbrages_and_veils = list() //A list of the minds of all umbrages and veils.
+
+/datum/game_mode/umbrage
+	name = "umbrage"
+	config_tag = "umbrage"
+	antag_flag = ROLE_UMBRAGE
+	required_players = 1 //25
+	required_enemies = 1
+	recommended_enemies = 3
+	enemy_minimum_age = 14
+	restricted_jobs = list("AI", "Cyborg", "Captain") //Notice that chaplains aren't here. Umbrages aren't magical or cultlike!
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain")
+	announce_span = "velvet"
+	announce_text = "Eldritch aberrations are trying to enslave the station!\n\
+	<span class='velvet'>Umbrages</span>: Dominate the will of the crew and ascend into a progenitor.\n\
+	<span class='notice'>Crew</span>: Protect your minds and nullify the umbrages before they take over."
+	var/list/initial_umbrages = list()
+
+/datum/game_mode/umbrage/pre_setup()
+	if(config.protect_roles_from_antagonist)
+		restricted_jobs += protected_jobs
+	if(config.protect_assistant_from_antagonist)
+		restricted_jobs += "Assistant"
+	var/starting_umbrages = Clamp(round(num_players() / 10), 1, 3) //At least 1 umbrage, but no more than 3
+	for(var/i = starting_umbrages, i > 0, i--)
+		var/datum/mind/new_umbrage = pick(antag_candidates)
+		initial_umbrages += new_umbrage
+		antag_candidates -= new_umbrage
+		modePlayer += new_umbrage
+		new_umbrage.special_role = "Umbrage"
+		new_umbrage.restricted_roles = restricted_jobs
+		log_game("[new_umbrage.key] (ckey) has been selected as an umbrage.")
+	return 1
+
+/datum/game_mode/umbrage/post_setup()
+	for(var/U in initial_umbrages)
+		var/datum/mind/umbrage_mind = U
+		var/mob/living/H = umbrage_mind.current
+		antag_umbrage(H, 1)
+		greet_umbrage(H)
+		equip_umbrage(H)
+	..()
+	return 1
+
+/datum/game_mode/proc/greet_umbrage(mob/living/U) //Announcement and direction to the tutorial
+	if(!U)
+		return
```

Comment:
I asked in coderbus a while back and supposedly returning null accomplished the same thing according to some of the maintainers there.